### PR TITLE
Consolidate on functional options

### PR DESCRIPTION
**Pull request by nickstenning**
This changes the signature of NewClient to only accept variadic Options, and to return an error if one occurs while setting up the client.

This leaves us with the internal implementation details of the client and its options fully encapsulated by the package, which should give us substantial freedom to extend the API in future while minimising breakage for users.

### CODE REVIEW

File: client.go
Comment (nickstenning) [inline client.go:1]: Is there any particular reason to do this? Go generally favours brevity and avoiding repetition in type signatures.

Comment (mattt) [reply client.go:1]: I think we'll want more than one option type in the library. For example, the signature of `CreatePrediction` after adding streaming support makes me think functional `CreatePredictionOptions` would be a good fit for that API as well https://github.com/replicate/replicate-go/pull/10#discussion_r1279920589.

Comment (nickstenning) [reply client.go:1]: Ok, fair enough!

**Review by mattt — APPROVED** [review]
Thanks again for your help with this, @nickstenning!

**Comment by mattt** [comment]
These changes are now available in [v0.6.0](https://github.com/replicate/replicate-go/releases/tag/v0.6.0).

**Pull request merged**